clause/where.go

```diff
@@ -166,19 +166,58 @@ type NotConditions struct {
 }
 
 func (not NotConditions) Build(builder Builder) {
-	if len(not.Exprs) > 1 {
-		builder.WriteByte('(')
+	anyNegationBuilder := false
+	for _, c := range not.Exprs {
+		if _, ok := c.(NegationExpressionBuilder); ok {
+			anyNegationBuilder = true
+			break
+		}
 	}
 
-	for idx, c := range not.Exprs {
-		if idx > 0 {
-			builder.WriteString(AndWithSpace)
+	if anyNegationBuilder {
```

Comment:
Do we need to distinguish negative expressions when not.Exprs length is greater than 1? I think the following statements should be the same?
```
	DB.Not(DB.Where("a=1").Where("b=1"))
	DB.Not(DB.Where("a=1").Where(clause.Eq{Column: "b", Value: 1}))
```

Reply:
I agree with your point of view, but implementing it that way would result in compatibility issues with the following code:
`DB.Not(&Table{A: 1, B: 1})`
This code, even in versions prior to v1.25.5, results in WHERE (a <> 1 AND b <> 2). Therefore, while your proposed logic seems correct, we need to consider backward compatibility with existing code implementations.

Reply:
I got it, can you add some test about this case ?